### ISSUE TITLE
[Bug] Standardization of the output and inverse transform of standard deviation

### ISSUE THREAD

**Issue by BidhanChandra** [cornellius-gp/gpytorch#2476]
# 🐛 Bug

Hi, 
I am trying to calculate the uncertainty at the predicted results. For my training, I am using StandardScaler on my output, which is performing much well rather than the unprocessed output.

y_scaler = StandardScaler().fit(targets_train_en)
y_train = torch.from_numpy(y_scaler.transform(targets_train_en)).float().squeeze()
y_valid = torch.from_numpy(y_scaler.transform(targets_test_en)).float().squeeze()

 While having predictions, I am doing inverse transform on my predicted data

model.eval()
likelihood.eval()

with torch.no_grad():
    observed_pred = likelihood(model(features_soap_test_gp))
    output = observed_pred.mean
    output_std = observed_pred.stddev

y_pred = y_scaler.inverse_transform(output.reshape(-1, 1))
y_std = y_scaler.inverse_transform(output_std.reshape(-1, 1))

While the predictions are good, but standard deviations are almost similar to predictions, which should not be the case. 
How to solve this problem?

**Comment by Balandat** [comment]
I am not familiar with `StandardScaler` but it seems like if that de-means the data that mean would be added to the std in this line: 
```
y_std = y_scaler.inverse_transform(output_std.reshape(-1, 1))
```
which is not what you want (you only want to scale the std by the scaling factor)

**Comment by BidhanChandra** [comment]
Thank you for your response. I understood your point. It should be just scaled and the mean of the original values should not be added. 
I have another question, does gpytorch have the same facility like **normalize_y** as in Gaussian process regression implementation of sklearn?

**Comment by Balandat** [comment]
I don't think GPyTorch has this functionality. BoTorch provides this as a transform though: https://github.com/pytorch/botorch/blob/main/botorch/models/transforms/outcome.py#L223
